src/ir_interpreter/interpret_ir.ml

```diff
@@ -822,7 +833,7 @@ and interpret_func env at sort x args f c v (k : V.value V.cont) =
 
 and interpret_message env at x args f c v (k : V.value V.cont) =
   let v_caller, v_reply, v_reject = match V.as_tup c with
-    | [v_caller; v_reply; v_reject] -> v_caller, v_reply, v_reject
+    | [v_caller; v_reply; v_reject; _v_cleanup] -> v_caller, v_reply, v_reject
```

Comment:
We _could_ save the cleanup in a global and run it when there is an error (e.g. `assert`ion). WDYT?

Reply:
Alternatively we could save it into the continuation environment and invoke it before fatal errors!

Reply:
This. But where should we keep the cleanup chain? In a global? Then we are stateful and have to be careful to reset it to `None` (or `@cleanup`) every time we send messages?

Reply:
I left a TODO in the description, maybe in another PR.